### PR TITLE
feat: [CDS-54454]: AWS Lambda Harness File Store

### PR DESCRIPTION
**Pull request by alluvamsi-harness**
## Describe your changes
Support Harness FileStore For Lambda specification
image

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44646)

### CODE REVIEW

**Review by rohitk-h — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44646)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Comment by rohitk-h** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java` line 357 at r1](https://reviewable.io/reviews/harness/harness-core/44646#-NPZksJi9hYSmAc-UFLK:-NPZksJi9hYSmAc-UFLL:b-w804ru) ([raw file](https://github.com/harness/harness-core/blob/f98e9091c02e9a45d704b9f989fef110d2c7a4e1/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java#L357)):*
> ```Java
>     TaskChainResponse taskChainResponse = null;
>     if (manifestOutcome.getStore() != null && ManifestStoreType.HARNESS.equals(manifestOutcome.getStore().getKind())) {
>       String manifestContent = fetchFilesContentFromLocalStore(ambiance, manifestOutcome, logCallback).get(0);
> ```

This function implementation seems to be missing.

**Review by rohitk-h — DISMISSED** [review]
> Reviewed 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44646)* status: 5 file(s) to review. 1 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java` line 890 at r2](https://reviewable.io/reviews/harness/harness-core/44646#-NPnzFjW2gCUNCDKyPFH:-NPnzFjW2gCUNCDKyPFI:b-68nexj) ([raw file](https://github.com/harness/harness-core/blob/9491f2a824fa2e53c94c49f9e48421ad5a7d5281/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java#L890)):*
> ```Java
>     for (ManifestOutcome manifest : ecsManifestsOutcomes) {
>       if (manifest.getStore() != null && ManifestStoreType.isInGitSubset(manifest.getStore().getKind())) {
>         isGitManifest = true;
> ```

suggestion: returning true from here itself is better instead of going through full list. 
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/beans/AwsLambdaHarnessStoreFilesResult.java` line 17 at r2](https://reviewable.io/reviews/harness/harness-core/44646#-NPo-39q79wBF1995-T4:-NPo-39q79wBF1995-T5:bqwdisf) ([raw file](https://github.com/harness/harness-core/blob/9491f2a824fa2e53c94c49f9e48421ad5a7d5281/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/beans/AwsLambdaHarnessStoreFilesResult.java#L17)):*
> ```Java
> @Builder
> public class AwsLambdaHarnessStoreFilesResult {
>   List<String> filesContent;
> ```

Can we add @Expression annotation the field?

**Review by piyush-bhuwalka — DISMISSED** [review]
> Reviewed 1 of 2 files at r1, 4 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44646)* status: :shipit: complete! ✓

**Review by alluvamsi-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44646)* status: 2 file(s) to review. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java` line 890 at r2](https://reviewable.io/reviews/harness/harness-core/44646#-NPnzFjW2gCUNCDKyPFH:-NPqRoiKBgpAka5ZjudM:b-896fjx) ([raw file](https://github.com/harness/harness-core/blob/9491f2a824fa2e53c94c49f9e48421ad5a7d5281/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/CDStepHelper.java#L890)):*
<details><summary><i>Previously, rohitk-h (Rohit Karelia) wrote…</i></summary><blockquote>

suggestion: returning true from here itself is better instead of going through full list. 
</blockquote></details>

Done
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/beans/AwsLambdaHarnessStoreFilesResult.java` line 17 at r2](https://reviewable.io/reviews/harness/harness-core/44646#-NPo-39q79wBF1995-T4:-NPqRrkd5tcVyZVPoUNd:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/9491f2a824fa2e53c94c49f9e48421ad5a7d5281/125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/beans/AwsLambdaHarnessStoreFilesResult.java#L17)):*
<details><summary><i>Previously, rohitk-h (Rohit Karelia) wrote…</i></summary><blockquote>

Can we add @Expression annotation the field? 
</blockquote></details>

Done

File: 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java [hunk 1]
```diff
@@ -382,11 +391,78 @@ public TaskChainResponse startChainLink(Ambiance ambiance, StepElementParameters
...
+          awsLambdaManifestOutcomeList, awsLambdaStepPassThroughDataBuilder);
+    } else {
+      taskChainResponse = prepareManifestHarnessStoreTask(
+          ambiance, stepElementParameters, infrastructureOutcome, awsLambdaStepPassThroughDataBuilder, logCallback);
+    }
+
+    return taskChainResponse;
+  }
+
+  private TaskChainResponse prepareManifestHarnessStoreTask(Ambiance ambiance, StepElementParameters stepParameters,
```
Comment (sainath-b) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:420]: can we pass passthrough data directly here instead of builder?

Comment (alluvamsi-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/aws/lambda/AwsLambdaHelper.java:420]: Done

**Comment by alluvamsi-harness** [comment]
trigger utAll

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 1 of 4 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44646)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by piyush-bhuwalka — APPROVED** [review]
> Reviewed 2 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44646)* status: 1 file(s) to review.

**Review by rohitk-h — APPROVED** [review]
> Reviewed 2 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44646)* status: :shipit: complete! ✓

**Pull request merged**